src/effects/VST3/VST3EffectsModule.cpp

```diff
@@ -0,0 +1,320 @@
+   }
+   wxDirTraverseResult OnDir(const wxString& dirname) override
+   {
+      if(dirname.Matches("*.vst3"))
+      {
+         mOnPluginFound(dirname);
+         return wxDIR_IGNORE;//do not look inside...
+      }
+      return wxDIR_CONTINUE;
+   }
+};
+
+const TranslatableString VST3EffectsModule::PluginType = XO("VST3");
+
+std::shared_ptr<VST3::Hosting::Module> VST3EffectsModule::GetModule(const wxString& path)
+{
+   const auto it = mModules.find(path);
+   if(it != mModules.end())
+   {
+      if(auto lock = it->second.lock())
+         return lock;
+   }
+
+   std::string moduleCreateError;
+   auto module = VST3::Hosting::Module::create(path.ToStdString(), moduleCreateError);
+   if(!module)
+      throw std::logic_error(moduleCreateError.c_str());
+
+   mModules[path] = module;
+   return module;
+}
+
+Steinberg::Vst::IHostApplication* VST3EffectsModule::LocalContext()
+{
+   static Steinberg::Vst::HostApplication localContext;
+   return &localContext;
+}
+
+PluginPath VST3EffectsModule::GetPath()
+{
+   return {};
+}
+
+ComponentInterfaceSymbol VST3EffectsModule::GetSymbol()
+{
+   return XO("VST3 Effects");
+}
+
+VendorSymbol VST3EffectsModule::GetVendor()
+{
+   return XO("The Audacity Team");
+}
+
+wxString VST3EffectsModule::GetVersion()
+{
+   return AUDACITY_VERSION_STRING;
+}
+
+TranslatableString VST3EffectsModule::GetDescription()
+{
+   return XO("Adds the ability to use VST3 effects in Audacity.");
+}
+
+bool VST3EffectsModule::Initialize()
+{
+   return true;
+}
+
+void VST3EffectsModule::Terminate()
+{
+}
+
+EffectFamilySymbol VST3EffectsModule::GetOptionalFamilySymbol()
+{
+   return PluginType;
+}
+
+const FileExtensions& VST3EffectsModule::GetFileExtensions()
+{
+   static const FileExtensions ext {
+      { _T("vst3") }
+   };
+   return ext;
+}
+
+FilePath VST3EffectsModule::InstallPath()
+{
+#ifdef VST3_DEFAULT_INSTALL_PATH
+   return FilePath { VST3_DEFUALT_INSTALL_PATH };
+#else
+   return {};
+#endif
+}
+
+bool VST3EffectsModule::AutoRegisterPlugins(PluginManagerInterface& pluginManager)
+{
+   return false;
+}
+
+PluginPaths VST3EffectsModule::FindPluginPaths(PluginManagerInterface& pluginManager)
+{
+   //Note: The host recursively scans these folders at startup in this order (User/Global/Application).
+   //https://developer.steinberg.help/display/VST/Plug-in+Locations
+
+   FilePaths pathList;
+#ifdef __WXMSW__
+   // Windows specific VST3 search path
+#ifdef __WIN64__
+   // x64 only
+   {
+      
+      auto length = GetEnvironmentVariableW(L"ProgramW6432", NULL, 0);
+      auto buf = std::make_unique<WCHAR[]>(length);
+      GetEnvironmentVariableW(L"ProgramW6432", buf.get(), length);
+
+      wxFileName path;
+      path.SetPath(static_cast<wchar_t*>(buf.get()));
+      path.AppendDir("Common Files");
+      path.AppendDir("VST3");
+      pathList.push_back(path.GetPath());
+   }
+#endif
+   // x64/x86
+   {
+      auto length = GetEnvironmentVariableW(L"ProgramFiles", NULL, 0);
```

Comment:
The same block of code except for the string literal.  I would make a lambda to avoid the duplication.